nixos/modules/services/networking/keepalived/default.nix

```diff
@@ -282,16 +295,25 @@ in
       };
     };
 
-    systemd.services.keepalived = {
+    systemd.services.keepalived = let
+      finalConfigFile = if cfg.secretFile == null then keepalivedConf else "/var/run/keepalived/keepalived.conf";
+    in {
       description = "Keepalive Daemon (LVS and VRRP)";
       after = [ "network.target" "network-online.target" "syslog.target" ];
       wants = [ "network-online.target" ];
       serviceConfig = {
         Type = "forking";
         PIDFile = pidFile;
         KillMode = "process";
+        RuntimeDirectory = "keepalived";
+        ExecStartPre = lib.optional (cfg.secretFile != null)
+        (pkgs.writeShellScript "pre-start" ''
+          umask 077
+          export $(xargs < ${cfg.secretFile})
```

Comment:
EnvironmentFile seems to be a good place to import these env vars.

Reply:
I think I got what you meant back then, instead of loading the content of `secretFile` with `export`, load it with `EnvironmentFile` but still replacing the config within the pre-start script.
I changed it that way now. Hope it's ok.